[[ Bug 22607 ]] Fix compilation errors pre-iOS 12

montegoulding opened the pull request:
This patch resolves a compilation error when building LiveCode against SDKs
that do not implement `UITraitCollection userInterfaceStyle`.

runrevmark commented on engine/src/mbliphonedc.mm:
```diff
@@ -859,16 +859,17 @@ static void MCScreenDCDoSnapshot(void *p_env)
 
 void MCScreenDC::getsystemappearance(MCSystemAppearance &r_appearance)
```
I take it `__IPHONE_12_0` is defined if it is the iOS13 SDK too?

It would be better to ifdef the whole method.

Also I don't think you need respondsToSelector with the ifdef.

montegoulding replied:
@runrevmark yes each new SDK defines one of these for this purpose and we have used these elsewhere in the engine. Don't we need `respondsToSelector` in the event the engine compiled against > iOS 12 is run on older OS?

What do you mean by `ifdef` the whole method. A completely separate `getsystemappearance` implementation?

runrevmark replied:
Doh! Of course... We can actually get away without the ifdef at all:
```
typedef int (*_userInterfaceStyle)(UITraitCollection *, SEL selector);
UITraitCollection *t_traits = [MCIPhoneGetRootView() traitCollection];
if ([t_traits respondsToSelector: @selector(userInterfaceStyle)] &&
        ((_userInterfaceStyle)objc_msgSend)(t_traits, @selector(userInterfaceStyle)) == 2)
...
```

livecodepanos commented:
Adding a comment so as I get notified when this is updated - so as ideally I merge it on time for the nightly build :)

montegoulding committed:
[[ Bug 22607 ]] Fix compilation errors pre-iOS 12

This patch resolves a compilation error when building LiveCode against SDKs
that do not implement `UITraitCollection userInterfaceStyle`.

runrevmark commented:
@livecode-vulcan review ok 1a53ce2

livecode-vulcan commented:
💙 review by @runrevmark ok 1a53ce28e0023e5b3cf84ff5736950c308fb8de5

livecode-vulcan commented:
😎 test success 1a53ce28e0023e5b3cf84ff5736950c308fb8de5

* try-community-armv6-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-sdk26_ndk16r15/builds/758)
* try-community-armv7-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv7-android-ndk16r15/builds/167)
* try-community-arm64-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-arm64-android-ndk16r15/builds/167)
* try-community-x86-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86-android-ndk16r15/builds/167)
* try-community-x86_64-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86_64-android-ndk16r15/builds/167)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/2284)
* try-community-universal-ios-iphoneos13.4: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos13.4/builds/18)
* try-community-universal-ios-iphonesimulator13.4: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator13.4/builds/18)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/2313)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/2300)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/2299)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/2334)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/2274)
